### PR TITLE
Problem: (fix #196) current goreleaser config may build binaries without ledger support

### LINKED ISSUE #196: Problem: current goreleaser config may build binaries without ledger support
https://github.com/crypto-com/chain-main/blob/master/.goreleaser.yml

to-be-confirmed / tested with rc0 -- https://github.com/crypto-com/chain-main/releases/tag/v0.7.0-rc0

### PR DESCRIPTION
**Pull request by linfeng-crypto**
solution: use docker to cross-compile the binary with `cgo` enabled

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -3,17 +3,49 @@ before:
   - go mod download
 
 builds:
-  - main: ./cmd/chain-maind
-    id: "chain-maind"
+  - id: "chain-maind-darwin"
```
Comment (yihuang) [inline .goreleaser.yml:6]: is this intended?

Comment (linfeng-crypto) [reply .goreleaser.yml:6]: I tried to use the same id, failed

**Review by leejw51crypto — APPROVED** [review]
lgtm

**Review by tomtau — COMMENTED** [review]
looks all right -- I don't have much experience with Go Releaser, so may wait for @allthatjazzleo or @calvinaco comments and merge it by today

**Comment by allthatjazzleo** [comment]
@tomtau I will try to release in fork repo

File: Makefile [hunk 2]
```diff
@@ -176,3 +179,34 @@ else
...
+	docker run \
+		-it \
+		--rm \
+		--privileged \
+		-e CGO_ENABLED=1 \
+		-v /var/run/docker.sock:/var/run/docker.sock \
+		-v `pwd`:/go/src/$(PACKAGE_NAME) \
+		-v ${GOPATH}/pkg:/go/pkg \
+		-w /go/src/$(PACKAGE_NAME) \
+		troian/golang-cross:${GOLANG_CROSS_VERSION} \
```
Comment (allthatjazzleo) [inline Makefile:194]: I got this error
```
docker: Error response from daemon: manifest for troian/golang-cross:1.13.15 not found: manifest unknown: manifest unknown.
```

Comment (tomtau) [reply Makefile:194]: @linfeng-crypto

File: .github/workflows/goreleaser.yml [hunk 3]
```diff
@@ -10,15 +10,11 @@ jobs:
...
       - uses: actions/checkout@v2
-      - name: Run GoReleaser
-        uses: goreleaser/goreleaser-action@v2
-        with:
-          version: latest
-          args: release --rm-dist
-        env:
-          GITHUB_TOKEN: ${{ secrets.RELEASE_GITHUB_TOKEN }}
+      - name: release dry run
+          run: make release-dry-run
```
Comment (allthatjazzleo) [inline .github/workflows/goreleaser.yml:15]:
```suggestion
        run: make release-dry-run
```
I got
```
Invalid workflow file
You have an error in your yaml syntax on line 15
```
should be related to indentation 
same for below

File: .goreleaser.yml [hunk 4]
```diff
@@ -28,7 +60,9 @@ archives:
     - goos: windows
       format: zip
   builds:
-  - chain-maind
+  - darwin
```
Comment (allthatjazzleo) [inline .goreleaser.yml:63]:
```suggestion
  - chain-maind-darwin
  - chain-maind-windows
  - chain-maind-linux
```
should match id

File: .github/workflows/goreleaser.yml [hunk 5]
```diff
@@ -10,15 +10,11 @@ jobs:
...
       - uses: actions/checkout@v2
-      - name: Run GoReleaser
-        uses: goreleaser/goreleaser-action@v2
-        with:
-          version: latest
-          args: release --rm-dist
-        env:
-          GITHUB_TOKEN: ${{ secrets.RELEASE_GITHUB_TOKEN }}
+      - name: release dry run
+         run: make release-dry-run
```
Comment (allthatjazzleo) [inline .github/workflows/goreleaser.yml:15]: the indent of `run` is still not aligned with `name`

**Review by allthatjazzleo — CHANGES_REQUESTED** [review]

File: Makefile [hunk 6]
```diff
@@ -176,3 +179,34 @@ else
 	rm -rf nix-remote-builder
 endif
 endif
+
+.PHONY: release-dry-run
+release-dry-run:
+	docker run \
+		-it \
```
Comment (allthatjazzleo) [inline Makefile:186]:
```suggestion
```
github actions cannot run with `-it`
with error
`the input device is not a TTY`

**Review by tomtau — APPROVED** [review]

**Review by allthatjazzleo — APPROVED** [review]

**Pull request merged**